src/app/shared/object-list/object-list.component.html

```diff
@@ -16,22 +16,46 @@
+                                             [selectionConfig]="selectionConfig"
+                                             (deselectObject)="deselectObject.emit($event)"
+                                             (selectObject)="selectObject.emit($event)"></ds-selectable-list-item-control>
+            <ds-importable-list-item-control *ngIf="importable" [object]="object"
+                                             [importConfig]="importConfig"
+                                             (importObject)="importObject.emit($event)"></ds-importable-list-item-control>
+            <ds-listable-object-component-loader [context]="context"
+                                                 [linkType]="linkType"
+                                                 [listID]="selectionConfig?.listId"
+                                                 [index]="i"
+                                                 [object]="object"
+                                                 [showThumbnails]="showThumbnails"
+                                                 [viewMode]="viewMode"
+                                                 (contentChange)="contentChange.emit($event)"></ds-listable-object-component-loader>
+          </li>
+        </ng-container>
+        <ng-container *ngIf="!listToPaginate">
+          <li *ngFor="let object of objects?.payload?.page; let i = index; let last = last" class="mt-4 mb-4 d-flex" [class.border-bottom]="hasBorder && !last" [attr.data-test]="'list-object' | dsBrowserOnly">
+            <ds-selectable-list-item-control *ngIf="selectable" [index]="i"
+                                             [object]="object"
+                                             [selectionConfig]="selectionConfig"
+                                             (deselectObject)="deselectObject.emit($event)"
+                                             (selectObject)="selectObject.emit($event)"></ds-selectable-list-item-control>
+            <ds-importable-list-item-control *ngIf="importable" [object]="object"
+                                             [importConfig]="importConfig"
+                                             (importObject)="importObject.emit($event)"></ds-importable-list-item-control>
+            <ds-listable-object-component-loader [context]="context"
+                                                 [linkType]="linkType"
+                                                 [listID]="selectionConfig?.listId"
+                                                 [index]="i"
+                                                 [object]="object"
+                                                 [showThumbnails]="showThumbnails"
+                                                 [viewMode]="viewMode"
+                                                 (contentChange)="contentChange.emit($event)"></ds-listable-object-component-loader>
+          </li>
+        </ng-container>
```

Comment:
Rather than duplicating this entire component that's used in almost every list in dspace to add that paginate pipe, to account for this single use case, I'd do the pagination on the objects before you pass them to ds-object-list